WIF updates (#7870)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/manage-access/set-up-bigquery-oauth.md

Workload Identity Federation (WIF) allows application workloads, running externally to <Constant name="cloud" />, to act as a service account without the need to manage service accounts or other keys for deployment environments. The following instructions will enable you to authenticate your BigQuery connection in <Constant name="cloud" /> using WIF. 
Currently, Microsoft Entra ID is the only supported identity provider (IdP). If you need additional IdP support, please contact your account team.

### 1. Set up Entra ID

Create an app in Entra where dbt will request access tokens when authenticating to BigQuery via the workload identity pool:

1. From the **app registrations** screen, click **New registration**.
2. Give the app a name that makes it easily identifiable.
3. Ensure **Supported account types** are set to “Accounts in this organizational directory only (Org name - Single Tenant).”
4. Click **Register** to see the application’s overview screen.

> 5. From the **app overview**, click **Expose an API** from the left menu.

Reviewer: ```suggestion
5. From the **app overview**, click **Expose an API** in the left menu.
```

6. Click **Add** next to Application ID URI. The field will automatically populate. 
7. Click **Save**.

> Workload Identity Federation utilizes a machine-to-machine OAuth flow that is unattended by the user; as such, a redirect URI won't need to be set for the application. Step three in this section is crucial because it determines the audience for tokens issued from the app and informs the workpool in GCP whether the calling application has permission to access the resources guarded by the workpool.

Reviewer: ```suggestion
Workload Identity Federation utilizes a machine-to-machine OAuth flow that is unattended by the user; as such, a redirect URI won't need to be set for the application. Step 3 in this section is crucial because it determines the audience for tokens issued from the app and informs the workpool in GCP whether the calling application has permission to access the resources guarded by the workpool.
```

- **Related documentation:** [GCP &mdash; Prepare your external identity provider](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#create)

### 2. Create a Workpool and Workpool Provider in GCP

> 1. In your GCP account's main menu, navigate to **IAM & Admin** and click the **Workload Identity Federation** option (not to be confused with the **Workforce Identity Federation** option directly adjacent). 

Reviewer: ```suggestion
1. In your GCP account's main menu, navigate to **IAM & Admin** and click the **Workload Identity Federation** option (not to be confused with the **Work_force_ Identity Federation** option directly adjacent). 
```

> 2. If you haven’t created a workpool yet, click **Get started** or create a new workpool (button near the top of the page).

Reviewer: ```suggestion
2. If you haven’t created a workpool yet, click **Get started** or create a new workpool (use button near the top of the page).
```

3. Give the workpool a name and description. Per the [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation#pools), a new pool should be created for each non-Google Cloud environment that needs to access Google Cloud resources, such as development, staging, or production environments. The workpool should be named accordingly to make it easily identifiable in the future.
4. When creating your provider:
    - Set the type of the provider to **OpenID Connect (OIDC)**.

>     - Name the provider something identifiable, like `Entra ID`.

Reviewer: ```suggestion
    - Name the provider to something identifiable, like `Entra ID`.
```

    - Set the URL to https://sts.windows.net/YOUR_TENANT_ID/. This can be found in the token itself, if you decode it via jwt.io. You can also see a reference to the expected issuer URL for Entra in the [GCP documentation for WIF](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#create_the_workload_identity_pool_and_provider).
        - The tenant (provider) ID can be found in the app registration created in [section 1 of these instructions](#1-set-up-entra-id); it's called **Directory (tenant) ID** and can be found in the overview section for the application.
    - For **Audiences**, select **Allowed Audiences** and set the value to the **Application ID URI** that was defined for your Entra ID app.
5. Click **Continue**.
6. Under **Configure provider attributes**, set the mapping for `google.subject` to `assertion.sub`.
7. Click **Save**.

### 3. Service Account Impersonation

> A workpool either uses a service account or is granted direct resource access to determine which resources a caller has access to. The [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#access) provides more detailed information on configuring both for your workpool. We chose the service account approach in our implementation because it offered greater flexibility.

Reviewer: ```suggestion
A workpool either uses a service account or is granted direct resource access to determine which resources a caller can access. The [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#access) provides more detailed information on configuring both for your workpool. We chose the service account approach in our implementation because it offered greater flexibility.
```

    - The Application ID URI should be set to the expected audience claim on tokens issued from the Entra application. It will be the same URI your workpool provider has been configured to expect.
    - You do not have to add the Redirect URI to your Entra application

> ### 4. Create connections in dbt

Reviewer: ```suggestion
### 5. Create connections in dbt
```

    - Service JSON 
        - You must create a separate connection with the Service JSON configuration.

> ### 5. Set up project

Reviewer: ```suggestion
### 6. Set up project
```

4. Select the **Connection** with the WIF configuration.
5. Configure the remainder of the project with the appropriate fields.

> ### 6. Set up deployment environment

Reviewer: ```suggestion
### 7. Set up deployment environment
```
